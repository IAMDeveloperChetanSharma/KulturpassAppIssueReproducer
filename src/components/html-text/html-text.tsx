import type { ChildNode, Element as ElementNode, Text as TextNode } from 'domhandler/lib/node'
import { ElementType, parseDocument } from 'htmlparser2'
import React, { useCallback } from 'react'
import { StyleProp, StyleSheet, Text, TextStyle } from 'react-native'
import { useTranslation } from '../../services/translation/translation'

import { openLink } from '../../utils/links/utils'

type InlineTextLinkProps = {
  text: string
  link: string
  textStyle?: StyleProp<TextStyle>
}
const InlineTextLink: React.FC<InlineTextLinkProps> = ({ text, link, textStyle }) => {
  const { t } = useTranslation()
  const handlePress = useCallback(() => openLink(link), [link])

  return (
    <Text
      style={[textStyle, styles.linkText]}
      accessible
      accessibilityLabel={text}
      accessibilityRole="link"
      accessibilityHint={t('external_link_short_accessibility_announcement')}
      onPress={handlePress}>
      {text}
    </Text>
  )
}

const sanitizeHtml = (text: string) => {
  return text.replace(/<br \/>/g, '\n').replace(/<br>/g, '\n')
}

type HtmlTextProps = {
  testID: string
  style: StyleProp<TextStyle>
  html: string
}

export const HtmlText: React.FC<HtmlTextProps> = ({ html, testID, style }) => {
  const renderTextNode = (textNode: TextNode, key: string, textStyle?: StyleProp<TextStyle>) => {
    const text = textNode.data
    return (
      <Text key={key} style={textStyle}>
        {text}
      </Text>
    )
  }

  const renderInlineLink = (textNode: TextNode, key: string, textStyle?: StyleProp<TextStyle>) => {
    const text = textNode.data
    let link: string = ''

    if (textNode.parent && textNode.parent.type === ElementType.Tag) {
      link = textNode.parent?.attribs?.href
    }

    return <InlineTextLink key={key} text={text} link={link} textStyle={textStyle} />
  }

  const renderElement = (element: ElementNode, key: string, textStyle?: StyleProp<TextStyle>) => {
    if (element.children.length === 1) {
      const elementKey = key + '-0'

      if (element.children[0].type === ElementType.Text) {
        switch (element.name) {
          case 'b':
            return renderTextNode(element.children[0], elementKey, [textStyle, styles.boldText])
          case 'i':
            return renderTextNode(element.children[0], elementKey, [textStyle, styles.italicText])
          case 'a':
            return renderInlineLink(element.children[0], elementKey + '-0', textStyle)
        }
      }
    }

    // nested text groups need a container
    return <Text key={key}>{element.children.map((c, i: number) => renderNode(c, `child-${i}`, textStyle))}</Text>
  }

  const renderNode = (node: ChildNode, index: string, textStyle?: StyleProp<TextStyle>) => {
    switch (node.type) {
      case ElementType.Text:
        return renderTextNode(node, index, textStyle)
      case ElementType.Tag:
        return renderElement(node, index, textStyle)
    }
    return null
  }

  const sanitizedHtml = sanitizeHtml(html)
  const document = parseDocument(sanitizedHtml)

  if (document.children.length === 0) {
    return (
      <Text testID={testID} style={style}>
        {sanitizedHtml}
      </Text>
    )
  }

  if (document.children.length === 1) {
    return renderNode(document.children[0], '0', style)
  }

  return (
    <Text testID={testID} style={style}>
      {document.children.map((c, i) => renderNode(c, 'sub' + i, style))}
    </Text>
  )
}

const styles = StyleSheet.create({
  boldText: { fontWeight: '700' },
  // TODO: extract textStyles below
  linkText: { textDecorationLine: 'underline' },
  italicText: { fontStyle: 'italic' },
})
