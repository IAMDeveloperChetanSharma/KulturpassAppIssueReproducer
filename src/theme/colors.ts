export const colors = {
  moonDarkest: '#0C0016',
  moonDarkest10: '#0C00161A',
  moonDarker: '#202325',
  moonDark: '#333333',
  moonBase: '#4D4D4D',
  moonLight: '#666666',
  moonLighter: '#808080',
  sunDark: '#999999',
  sunBase: '#B3B3B3',
  sunLight: '#CCCCCC',
  sunLighter: '#E6E6E6',
  sunLightest: '#F2F2F2',
  primaryDark: '#090A0A',
  primaryDarkest: '#4C56D7',
  primaryBase: '#0059F5',
  primaryLight: '#767FF9',
  primaryLighter: '#ABA8FC',
  primaryLightest: '#D7D3FE',
  secondaryDarkest: '#f12aae',
  secondaryBase: '#F566C5',
  secondaryLight: '#FB90D3',
  secondaryLighter: '#FFB7E2',
  secondaryLightest: '#FFDBF0',
  redDarkest: '#bf0000',
  redBase: '#FF0000',
  redLight: '#FF6846',
  redLighter: '#FF9E81',
  redLightest: '#FFCFBF',
  greenDarkest: '#10a98d',
  greenBase: '#15D7B4',
  greenLight: '#73E2C6',
  greenLighter: '#A7ECD9',
  greenLightest: '#D4F6EC',
  yellowDarkest: '#E9C30C',
  yellowBase: '#F5D640',
  yellowLight: '#FCE075',
  yellowLighter: '#FFEAA4',
  yellowLightest: '#FFF4D2',
  basicWhite: '#FFFFFF',
  basicBlack: '#0C0016',
  basicBackground: '#F3F3F7',

  transparentPrimaryLight50: '#367fff80',
  transparentPrimaryLight20: '#367fff33',
  transparentPrimaryLightest50: '#D7D3FE80',
  transparentBlack30: '#0C00161E',
  transparentBlack80: '#0C0016CC',
  transparentBlack70: '#0C0016B3',
  transparentBlack45: '#0C00162D',
  transparentBlack40: '#0C001666',
  transparentBlack60: '#0C00163C',
  transparentWhite100: '#FFFFFF00',
  transparentWhite45: '#FFFFFF2D',
  transparentWhite50: '#FFFFFF80',
} as const
