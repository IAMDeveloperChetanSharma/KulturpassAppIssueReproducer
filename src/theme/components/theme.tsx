import React, { PropsWithChildren, useEffect } from 'react'
import SystemNavigationBar from 'react-native-system-navigation-bar'
import { useTheme } from '../hooks/use-theme'

export const Theme: React.FC<PropsWithChildren> = ({ children }) => {
  const { colorScheme, colors } = useTheme()

  useEffect(() => {
    SystemNavigationBar.setNavigationColor(
      colors.secondaryBackground,
      colorScheme === 'light' ? 'dark' : 'light',
      'both',
    )
  }, [colorScheme, colors.secondaryBackground])

  return <>{children}</>
}
