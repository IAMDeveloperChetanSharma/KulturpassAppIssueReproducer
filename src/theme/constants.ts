export const HITSLOP_FAVORITES_LIST_ITEM = 5
export const HITSLOP = 12
export const HITSLOP_PIN = 10
