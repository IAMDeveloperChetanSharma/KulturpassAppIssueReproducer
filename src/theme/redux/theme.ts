import { createSlice, PayloadAction } from '@reduxjs/toolkit'

export type ThemeState = {
  darkThemePreviewEnabled: boolean
  forcedTheme: 'light' | 'dark' | null
}

export const themeInitialState: ThemeState = {
  darkThemePreviewEnabled: false,
  forcedTheme: null,
}

export const themeSlice = createSlice({
  name: 'theme',
  initialState: themeInitialState,
  reducers: {
    setDarkThemePreviewEnabled: (state, action: PayloadAction<boolean>) => {
      state.darkThemePreviewEnabled = action.payload
    },
    setForcedTheme: (state, action: PayloadAction<ThemeState['forcedTheme']>) => {
      state.forcedTheme = action.payload
    },
  },
})

export const { setDarkThemePreviewEnabled, setForcedTheme } = themeSlice.actions
