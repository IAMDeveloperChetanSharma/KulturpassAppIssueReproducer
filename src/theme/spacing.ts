export const spacing = [
  2, // 0
  4, // 1
  8, // 2
  10, // 3
  12, // 4
  16, // 5
  24, // 6
  32, // 7
  40, // 8
  48, // 9
  56, // 10
  64, // 11
  72, // 12
  80, // 13
] as const
