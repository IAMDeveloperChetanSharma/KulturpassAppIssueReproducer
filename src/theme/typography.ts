import { StyleSheet, TextStyle } from 'react-native'

export const typography = {
  primary: 'Nunito',
}

export const transformToBolderFontWeight = (input: TextStyle['fontWeight']): TextStyle['fontWeight'] => {
  switch (input) {
    case '100':
      return '300'
    case '200':
      return '400'
    case '300':
      return '500'
    case '400':
      return '600'
    case '500':
      return '700'
    case '600':
      return '800'
    case '700':
      return '900'
    case '900':
    case '800':
      return '900'
    case 'normal':
    default:
      return 'bold'
  }
}

export const textStyles = StyleSheet.create({
  /**
   * new design
   */
  HeadlineH1Black: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '900',
    fontSize: 40,
    lineHeight: 48,
  },
  HeadlineH2Black: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '900',
    fontSize: 32,
    lineHeight: 36,
  },
  HeadlineH3Extrabold: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '800',
    fontSize: 25,
    lineHeight: 29,
  },
  HeadlineH4Extrabold: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '800',
    fontSize: 21,
    lineHeight: 28,
  },
  HeadlineH4Bold: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '700',
    fontSize: 21,
    lineHeight: 28,
  },
  HeadlineH4Regular: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '400',
    fontSize: 21,
    lineHeight: 28,
  },

  SubtitleBlack: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '900',
    fontSize: 19,
    lineHeight: 23,
  },
  SubtitleExtrabold: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '800',
    fontSize: 19,
    lineHeight: 23,
  },
  SubtitleSemibold: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '800',
    fontSize: 19,
    lineHeight: 23,
  },
  SubtitleRegular: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '400',
    fontSize: 19,
    lineHeight: 23,
  },

  BodyBlack: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '900',
    fontSize: 17,
    lineHeight: 23,
  },
  BodyExtrabold: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '800',
    fontSize: 17,
    lineHeight: 23,
  },
  BodyBold: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '700',
    fontSize: 17,
    lineHeight: 23,
  },
  BodyMedium: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '700',
    fontSize: 17,
    lineHeight: 23,
  },
  BodyRegular: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '400',
    fontSize: 17,
    lineHeight: 23,
  },

  BodySmallExtrabold: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '800',
    fontSize: 14.8,
    lineHeight: 20,
    letterSpacing: 0.25,
  },
  BodySmallBold: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '700',
    fontSize: 14.8,
    lineHeight: 18,
    letterSpacing: 0.25,
  },
  BodySmallSemibold: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '600',
    fontSize: 14.8,
    lineHeight: 20,
    letterSpacing: 0.25,
  },
  BodySmallMedium: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '600',
    fontSize: 14.8,
    lineHeight: 18,
    letterSpacing: 0.25,
  },
  BodySmallRegular: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '400',
    fontSize: 14.8,
    lineHeight: 20,
    letterSpacing: 0.25,
  },

  CaptionExtrabold: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '800',
    fontSize: 13,
    lineHeight: 15,
    letterSpacing: 0.4,
  },
  CaptionSemibold: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '600',
    fontSize: 13,
    lineHeight: 15,
    letterSpacing: 0.4,
  },

  MicroExtraboldCaps: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '800',
    fontSize: 11,
    lineHeight: 14,
    letterSpacing: 1.4,
    textTransform: 'uppercase',
  },
  MicroExtrabold: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '800',
    fontSize: 11,
    lineHeight: 14,
    letterSpacing: 0.3,
  },
  MicroMedium: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '600',
    fontSize: 11,
    lineHeight: 12,
    letterSpacing: 0.4,
  },
  MicroMediumCaps: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '800',
    fontSize: 11,
    lineHeight: 14,
    letterSpacing: 1.5,
    textTransform: 'uppercase',
  },

  BodyPrimary1Dark: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '700',
    fontSize: 13,
    lineHeight: 16,
    letterSpacing: 0.4,
  },
  SuggestionInfo: {
    fontFamily: typography.primary,
    fontStyle: 'normal',
    fontWeight: '400',
    fontSize: 13,
    lineHeight: 19.5,
    letterSpacing: 0.4,
  },
})

export const textHighlighting = StyleSheet.create({
  Bold: {
    fontWeight: '700',
  },
  Link: {
    textDecorationLine: 'underline',
  },
  Italic: {
    fontStyle: 'italic',
  },
})
