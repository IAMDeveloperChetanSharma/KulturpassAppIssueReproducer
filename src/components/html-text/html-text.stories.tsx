import { ComponentStory, ComponentMeta } from '@storybook/react-native'
import React from 'react'
import { ScrollView } from 'react-native'
import { colors } from '../../theme/colors'
import { textStyles } from '../../theme/typography'
import { HtmlText } from './html-text'

const basicHtml = `
<b>bold text</b>
<i>italic text</i>
regular linebreak
regular linebreak next line
br linebreak<br />br linebreak next line
<a href="http://www.sap.de">link to sap.de</a>
`

const productDescriptionExample1 = `
Ein Erlebnis wie kein anderes. Im Kino mit Popcorn schaut es sich einfach am Besten!

Wir bieten verschiedene Preiskategorien an:

<b>Loge</b>
<i>Unser bequemster Platz</i>

<b>Parkett</b>
<i>Preiswerteste Wahl</i>

Mehr Infos <a href='https://www.sap.de'>hier</a>!`

const productDescriptionExample2 = `<p><strong>Peter Orloff &amp; Schwarzmeer Kosaken-Chor</strong><br />
Es gibt viele Kosakenchöre, aber nur einen Schwarzmeer – Kosakenchor urteilte die WAZ. Das Konzert ist eine musikalische Reise durch die märchenhafte Welt des alten Russland mit Romanzen, Geschichten und Balladen von grandioser Ausdruckskraft, tiefer Melancholie und überschäumendem Temperament.<br />
<br />
Der berühmte Schwarzmeer Kosaken-Chor unter der musikalischen Gesamtleitung und persönlicher Mitwirkung von Peter Orloff, der einst vor über 50 Jahren als jüngster Sänger aller Kosakenchöre der Welt seine legendäre Karriere begründete, ist wieder auf Europa-Tournee.<br />
<br />
<strong>Wertcode für ein Touch Konzert</strong><br />
<br />
<a href="https://www.touch-band.rocks" rel="nofollow noreferrer" target="_blank">Https Link 1</a><br />
Line1<br />
<br />
Line2<br />
<a href="https://www.touch-band.rocks" rel="nofollow noreferrer" target="_blank">Https Link 2</a><br />
<br />
<em>Line3</em><br />
<a href="https://www.touch-band.rocks" rel="nofollow noreferrer" target="_blank">Https Link</a><br />
<br />
<br />
<strong>Test</strong></p>`

const componentMeta: ComponentMeta<typeof HtmlText> = {
  title: 'HtmlText',
  component: HtmlText,
  args: {
    testID: 'some-test-id',
    style: { ...textStyles.BodyRegular, color: colors.moonDarkest },
    html: basicHtml,
  },
}

export default componentMeta

export const Basic: ComponentStory<typeof HtmlText> = args => (
  <ScrollView>
    <HtmlText {...args} />
  </ScrollView>
)
export const ProductDescription1: ComponentStory<typeof HtmlText> = args => (
  <ScrollView>
    <HtmlText {...args} html={productDescriptionExample1} />
  </ScrollView>
)
export const ProductDescription2: ComponentStory<typeof HtmlText> = args => (
  <ScrollView>
    <HtmlText {...args} html={productDescriptionExample2} />
  </ScrollView>
)
