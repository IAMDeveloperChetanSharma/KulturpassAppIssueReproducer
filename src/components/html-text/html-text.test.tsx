import { render, screen } from '@testing-library/react-native'
import React from 'react'
import { colors } from '../../theme/colors'
import { textStyles } from '../../theme/typography'
import { HtmlText } from './html-text'

const renderHtmlText = (html: string) => {
  render(
    <HtmlText testID={'some-test-ID'} style={{ ...textStyles.BodyRegular, color: colors.moonDarkest }} html={html} />,
  )
}

describe('HtmlText', () => {
  it('returns a pure text string wrapped into a text element', () => {
    const input = 'hello world'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
  it('replaces a <b> tag', () => {
    const input = '<b>bold</b>'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
  it('replaces a <b> tag with preluding text', () => {
    const input = 'hello <b>bold world</b>'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
  it('replaces a <i> tag', () => {
    const input = '<i>italic</i>'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
  it('replaces a <i> tag with preluding text', () => {
    const input = 'hello <i>italic world</i>'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
  it('replaces html entities', () => {
    const input = 'hell&ouml;&auml;&szlig;'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
  it('replaces all html entities', () => {
    const input = 'hell&ouml;&auml;&szlig;Lgt;&gt;'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
  it('replaces &lgt; &gt; with < and >', () => {
    const input = 'Bigger &gt; smaller &lt;'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
  it('replaces links which are opened in an external browser', () => {
    const input = '<a href="http://www.somewhere.de/somelink">linktext</a>'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
  it('filters out any other html', () => {
    const input = 'hack attack: <script>/* hack attack *?</script>'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
  it('replaces a <br /> with a linkebreak', () => {
    const input = 'line 1<br />line2'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
  it('keeps \n linkebreaks', () => {
    const input = 'line 1\nline2'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
  it('replaces a complex product description', () => {
    const input = `
      Ein Erlebnis wie kein anderes. Im Kino mit Popcorn schaut es sich einfach am Besten!

      Wir bieten verschiedene Preiskategorien an:

      <b>Loge</b>
      <i>Unser bequemster Platz</i>

      <b>Parkett</b>
      <i>Preiswerteste Wahl</i>

      Mehr Infos <a href='https://www.sap.de'>hier</a>!`
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('ignores simple unclosed tags', () => {
    const input = 'Hello <b world'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
  it('doesnt throw an error and still renders something if html invalid', () => {
    const input = 'Hello <b wo<script>rld<i><script italic</i>'
    renderHtmlText(input)
    expect(screen.toJSON()).toMatchSnapshot()
  })
})
