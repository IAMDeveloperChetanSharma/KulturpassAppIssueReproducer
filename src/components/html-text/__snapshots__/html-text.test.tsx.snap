// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HtmlText doesnt throw an error and still renders something if html invalid 1`] = `
<Text
  style={
    {
      "color": "#0C0016",
      "fontFamily": "Nunito",
      "fontSize": 17,
      "fontStyle": "normal",
      "fontWeight": "400",
      "lineHeight": 23,
    }
  }
  testID="some-test-ID"
>
  <Text
    style={
      {
        "color": "#0C0016",
        "fontFamily": "Nunito",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 23,
      }
    }
  >
    Hello 
  </Text>
  <Text>
    <Text
      style={
        {
          "color": "#0C0016",
          "fontFamily": "Nunito",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "400",
          "lineHeight": 23,
        }
      }
    >
      rld
    </Text>
    <Text
      style={
        [
          {
            "color": "#0C0016",
            "fontFamily": "Nunito",
            "fontSize": 17,
            "fontStyle": "normal",
            "fontWeight": "400",
            "lineHeight": 23,
          },
          {
            "fontStyle": "italic",
          },
        ]
      }
    />
  </Text>
</Text>
`;

exports[`HtmlText filters out any other html 1`] = `
<Text
  style={
    {
      "color": "#0C0016",
      "fontFamily": "Nunito",
      "fontSize": 17,
      "fontStyle": "normal",
      "fontWeight": "400",
      "lineHeight": 23,
    }
  }
  testID="some-test-ID"
>
  <Text
    style={
      {
        "color": "#0C0016",
        "fontFamily": "Nunito",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 23,
      }
    }
  >
    hack attack: 
  </Text>
</Text>
`;

exports[`HtmlText ignores simple unclosed tags 1`] = `
<Text
  style={
    {
      "color": "#0C0016",
      "fontFamily": "Nunito",
      "fontSize": 17,
      "fontStyle": "normal",
      "fontWeight": "400",
      "lineHeight": 23,
    }
  }
>
  Hello 
</Text>
`;

exports[`HtmlText keeps 
 linkebreaks 1`] = `
<Text
  style={
    {
      "color": "#0C0016",
      "fontFamily": "Nunito",
      "fontSize": 17,
      "fontStyle": "normal",
      "fontWeight": "400",
      "lineHeight": 23,
    }
  }
>
  line 1
line2
</Text>
`;

exports[`HtmlText replaces &lgt; &gt; with < and > 1`] = `
<Text
  style={
    {
      "color": "#0C0016",
      "fontFamily": "Nunito",
      "fontSize": 17,
      "fontStyle": "normal",
      "fontWeight": "400",
      "lineHeight": 23,
    }
  }
>
  Bigger &gt; smaller &lt;
</Text>
`;

exports[`HtmlText replaces a <b> tag 1`] = `
<Text
  style={
    [
      {
        "color": "#0C0016",
        "fontFamily": "Nunito",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 23,
      },
      {
        "fontWeight": "700",
      },
    ]
  }
>
  bold
</Text>
`;

exports[`HtmlText replaces a <b> tag with preluding text 1`] = `
<Text
  style={
    {
      "color": "#0C0016",
      "fontFamily": "Nunito",
      "fontSize": 17,
      "fontStyle": "normal",
      "fontWeight": "400",
      "lineHeight": 23,
    }
  }
  testID="some-test-ID"
>
  <Text
    style={
      {
        "color": "#0C0016",
        "fontFamily": "Nunito",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 23,
      }
    }
  >
    hello 
  </Text>
  <Text
    style={
      [
        {
          "color": "#0C0016",
          "fontFamily": "Nunito",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "400",
          "lineHeight": 23,
        },
        {
          "fontWeight": "700",
        },
      ]
    }
  >
    bold world
  </Text>
</Text>
`;

exports[`HtmlText replaces a <br /> with a linkebreak 1`] = `
<Text
  style={
    {
      "color": "#0C0016",
      "fontFamily": "Nunito",
      "fontSize": 17,
      "fontStyle": "normal",
      "fontWeight": "400",
      "lineHeight": 23,
    }
  }
>
  line 1
line2
</Text>
`;

exports[`HtmlText replaces a <i> tag 1`] = `
<Text
  style={
    [
      {
        "color": "#0C0016",
        "fontFamily": "Nunito",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 23,
      },
      {
        "fontStyle": "italic",
      },
    ]
  }
>
  italic
</Text>
`;

exports[`HtmlText replaces a <i> tag with preluding text 1`] = `
<Text
  style={
    {
      "color": "#0C0016",
      "fontFamily": "Nunito",
      "fontSize": 17,
      "fontStyle": "normal",
      "fontWeight": "400",
      "lineHeight": 23,
    }
  }
  testID="some-test-ID"
>
  <Text
    style={
      {
        "color": "#0C0016",
        "fontFamily": "Nunito",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 23,
      }
    }
  >
    hello 
  </Text>
  <Text
    style={
      [
        {
          "color": "#0C0016",
          "fontFamily": "Nunito",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "400",
          "lineHeight": 23,
        },
        {
          "fontStyle": "italic",
        },
      ]
    }
  >
    italic world
  </Text>
</Text>
`;

exports[`HtmlText replaces a complex product description 1`] = `
<Text
  style={
    {
      "color": "#0C0016",
      "fontFamily": "Nunito",
      "fontSize": 17,
      "fontStyle": "normal",
      "fontWeight": "400",
      "lineHeight": 23,
    }
  }
  testID="some-test-ID"
>
  <Text
    style={
      {
        "color": "#0C0016",
        "fontFamily": "Nunito",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 23,
      }
    }
  >
    
      Ein Erlebnis wie kein anderes. Im Kino mit Popcorn schaut es sich einfach am Besten!

      Wir bieten verschiedene Preiskategorien an:

      
  </Text>
  <Text
    style={
      [
        {
          "color": "#0C0016",
          "fontFamily": "Nunito",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "400",
          "lineHeight": 23,
        },
        {
          "fontWeight": "700",
        },
      ]
    }
  >
    Loge
  </Text>
  <Text
    style={
      {
        "color": "#0C0016",
        "fontFamily": "Nunito",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 23,
      }
    }
  >
    
      
  </Text>
  <Text
    style={
      [
        {
          "color": "#0C0016",
          "fontFamily": "Nunito",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "400",
          "lineHeight": 23,
        },
        {
          "fontStyle": "italic",
        },
      ]
    }
  >
    Unser bequemster Platz
  </Text>
  <Text
    style={
      {
        "color": "#0C0016",
        "fontFamily": "Nunito",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 23,
      }
    }
  >
    

      
  </Text>
  <Text
    style={
      [
        {
          "color": "#0C0016",
          "fontFamily": "Nunito",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "400",
          "lineHeight": 23,
        },
        {
          "fontWeight": "700",
        },
      ]
    }
  >
    Parkett
  </Text>
  <Text
    style={
      {
        "color": "#0C0016",
        "fontFamily": "Nunito",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 23,
      }
    }
  >
    
      
  </Text>
  <Text
    style={
      [
        {
          "color": "#0C0016",
          "fontFamily": "Nunito",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "400",
          "lineHeight": 23,
        },
        {
          "fontStyle": "italic",
        },
      ]
    }
  >
    Preiswerteste Wahl
  </Text>
  <Text
    style={
      {
        "color": "#0C0016",
        "fontFamily": "Nunito",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 23,
      }
    }
  >
    

      Mehr Infos 
  </Text>
  <Text
    accessibilityLabel="hier"
    accessibilityRole="link"
    accessible={true}
    onPress={[Function]}
    style={
      [
        {
          "color": "#0C0016",
          "fontFamily": "Nunito",
          "fontSize": 17,
          "fontStyle": "normal",
          "fontWeight": "400",
          "lineHeight": 23,
        },
        {
          "textDecorationLine": "underline",
        },
      ]
    }
  >
    hier
  </Text>
  <Text
    style={
      {
        "color": "#0C0016",
        "fontFamily": "Nunito",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 23,
      }
    }
  >
    !
  </Text>
</Text>
`;

exports[`HtmlText replaces all html entities 1`] = `
<Text
  style={
    {
      "color": "#0C0016",
      "fontFamily": "Nunito",
      "fontSize": 17,
      "fontStyle": "normal",
      "fontWeight": "400",
      "lineHeight": 23,
    }
  }
>
  hellöäßLgt;&gt;
</Text>
`;

exports[`HtmlText replaces html entities 1`] = `
<Text
  style={
    {
      "color": "#0C0016",
      "fontFamily": "Nunito",
      "fontSize": 17,
      "fontStyle": "normal",
      "fontWeight": "400",
      "lineHeight": 23,
    }
  }
>
  hellöäß
</Text>
`;

exports[`HtmlText replaces links which are opened in an external browser 1`] = `
<Text
  accessibilityLabel="linktext"
  accessibilityRole="link"
  accessible={true}
  onPress={[Function]}
  style={
    [
      {
        "color": "#0C0016",
        "fontFamily": "Nunito",
        "fontSize": 17,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 23,
      },
      {
        "textDecorationLine": "underline",
      },
    ]
  }
>
  linktext
</Text>
`;

exports[`HtmlText returns a pure text string wrapped into a text element 1`] = `
<Text
  style={
    {
      "color": "#0C0016",
      "fontFamily": "Nunito",
      "fontSize": 17,
      "fontStyle": "normal",
      "fontWeight": "400",
      "lineHeight": 23,
    }
  }
>
  hello world
</Text>
`;
